src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/RuleRunnerFactory.cs

```diff
@@ -10,47 +10,41 @@ namespace Microsoft.DotNet.ApiCompatibility.Rules
 {
     public class RuleRunnerFactory
     {
-        private readonly bool _strictMode;
         private readonly string _leftName;
         private readonly string[] _rightNames;
         private readonly IEqualityComparer<ISymbol> _equalityComparer;
         private readonly bool _includeInternalSymbols;
+        private readonly bool _strictMode;
         private readonly bool _withReferences;
-        private RuleRunner _runner;
+        private readonly Lazy<RuleRunner> _runner;
 
-        public RuleRunnerFactory(string leftName, string[] rightNames, IEqualityComparer<ISymbol> equalityComparer, bool includeInternalSymbols, bool strictMode, bool withReferences)
+        public RuleRunnerFactory(string? leftName, string[]? rightNames, IEqualityComparer<ISymbol> equalityComparer, bool includeInternalSymbols, bool strictMode, bool withReferences)
         {
-            _strictMode = strictMode;
-            _withReferences = withReferences;
-            _equalityComparer = equalityComparer;
-            _includeInternalSymbols = includeInternalSymbols;
-            _leftName = string.IsNullOrEmpty(leftName) ? RuleRunner.DEFAULT_LEFT_NAME : leftName;
+            _leftName = string.IsNullOrEmpty(leftName) ? RuleRunner.DEFAULT_LEFT_NAME : leftName!;
             _rightNames = rightNames ?? new string[] { RuleRunner.DEFAULT_RIGHT_NAME };
-            if (_rightNames.Length <= 0)
+            if (_rightNames.Length == 0)
             {
-                throw new ArgumentException(nameof(rightNames), Resources.RightNamesAtLeastOne);
+                throw new ArgumentException(Resources.RightNamesAtLeastOne, nameof(rightNames));
```

Comment:
Fixing the order of the parameters passed in to the ArgumentException: `ArgumentException(string message, string paramName)` instead of `ArgumentException(string paramName, string message)`.